### PR TITLE
Maths validation bring in cellml xml namespace

### PR DESCRIPTION
**Pull request by hsorby**
Using a text based method to bring in the cellml xml namespace (if required) into math sub documents on validation.

Addresses issue #175.

### CODE REVIEW

**Comment by hsorby** [comment]
This pull request is ready for review.

File: src/validator.cpp [hunk 1]
```diff
@@ -16,22 +16,22 @@ limitations under the License.
...
 #include "namespaces.h"
 #include "utilities.h"
 #include "xmldoc.h"
 
-#include <algorithm>
-#include <cmath>
-#include <stdexcept>
-
-#include <libxml/uri.h>
-
```
Comment (agarny) [inline src/validator.cpp:34]: Why reordering our header files using an order that is different from what was done in a recent suggested reordering?! So much inconsistency.

**Review by agarny — APPROVED** [review]
This looks good to me.

File: src/validator.cpp (hunk 1 above)
Comment (hsorby) [reply src/validator.cpp:34]: I thought this was the consistent ordering defined by the style guide.  Associated header first, system headers, project headers.

Comment (agarny) [reply src/validator.cpp:34]: Am giving up. Will see that as part of issue #402.

File: src/validator.cpp [hunk 2]
```diff
@@ -835,9 +835,20 @@ void Validator::ValidatorImpl::validateWhen(const WhenPtr &when, const ResetPtr
 
 void Validator::ValidatorImpl::validateMath(const std::string &input, const ComponentPtr &component)
 {
+    std::string modifiedInput = input;
+
+    // It may be that we need to copy over the cellml namespace from the enclosing document.
+    auto cellmlNamespaceText = " xmlns:cellml=\"" + std::string(CELLML_2_0_NS);
+    if (input.find("cellml:units") != std::string::npos && input.find(cellmlNamespaceText) == std::string::npos) {
```
Comment (nickerso) [inline src/validator.cpp:842]: This will only work if we assume a `cellml` prefix is used.

**Review by nickerso — APPROVED** [review]
I think this works as well as our current handling of MathML and namespaces in parsing. So I think this improves the code, but needs further work to address these issues. See also https://github.com/cellml/libcellml/issues/118#issuecomment-235479335

**Pull request merged**